### ISSUE TITLE
Unzip with 7z big file compressed with ZIP_CM_XZ

### ISSUE THREAD

**Issue by F4r3n** [nih-at/libzip#241]
**Describe the Bug**
Whenever I zip a file with a size >50Mo with the compression mode XZ I cannot unzip it with 7z.
It works fine with small files.

**Expected Behavior**
Unzip a file zipped with the compression XZ by 7z.

**Observed Behavior**
When 7z tries to unzip it I get a "Data error"

**To Reproduce**
Set the compression mode as ZIP_CM_XZ and zip a file with a size >50Mo

**libzip Version**
Master (^1.7.3)

**Operating System**
Windows/Visual studio 2019

**Note**
LZMA works fine

**Comment by 0-wiz-0** [comment]
I just tried reproducing this (with ziptool from libzip git) but could not.
````
# ziptool foo.zip add_file 70mbfile  70mbfile 0 0 set_file_compression 0 xz 0
# 7z t foo.zip 

7-Zip (a) [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=de_AT.UTF-8,Utf16=on,HugeFiles=on,64 bits,32 CPUs x64)

Scanning the drive for archives:
1 file, 71031518 bytes (68 MiB)

Testing archive: foo.zip
--
Path = foo.zip
Type = zip
Physical Size = 71031518

Everything is Ok                                         

Size:       71079151
Compressed: 71031518
```
Can you perhaps share your data?

**Comment by F4r3n** [comment]
Hello,
I have tried:
 ziptool foo.zip add_file bigfile.txt bigfile.txt 0 0 set_file_compression 0 xz 0
on Ubuntu it works fine. 
But whenever I try to unzip the zip created under Windows I have an error.

Here is my file:
https://we.tl/t-0PDBI5V8vI

**Comment by 0-wiz-0** [comment]
Which tool do you use on Windows?
Have you asked the author of the Windows tool if they know what the problem is?

**Comment by F4r3n** [comment]
Under windows:
![7z](https://user-images.githubusercontent.com/6674322/121032987-1674fc80-c7ac-11eb-86fe-81afa469efdd.png)

**Comment by 0-wiz-0** [comment]
I've filed a bug report with 7zip:
https://sourceforge.net/p/sevenzip/discussion/45797/thread/73a6311ba1/

**Comment by F4r3n** [comment]
You managed to reproduce ?

**Comment by jloqfjgk** [comment]
For 7-Zip 21.02 alpha (Linux native version built from [official source](https://sourceforge.net/p/sevenzip/discussion/45797/thread/7fe6c21efa/)), I can unzip the archive created with libzip and xz compression (which contains `bigfile.txt`).

I haven't tested on Windows yet but you may want to try and see if you can reproduce the bug on alpha version.

**Comment by jloqfjgk** [comment]
Update: Igor Pavlov [pointed out](https://sourceforge.net/p/sevenzip/discussion/45797/thread/73a6311ba1/#63ac) that the bug has already been fixed in 21.01 alpha.

**Comment by F4r3n** [comment]
With the 21.0.2 it works fine, no more issue. Thanks
